crates/net/dns/src/lib.rs

```diff
@@ -402,7 +402,7 @@ fn convert_enr_node_record(enr: &Enr<SecretKey>) -> Option<DnsNodeRecordUpdate>
     }
     .into_ipv4_mapped();
 
-    let maybe_fork_id = enr.get(b"eth")?;
+    let maybe_fork_id = enr.get_raw_rlp(b"eth")?;
```

Comment:
can you please test this against the actual DNS records,

when I debugged this, this did not decode the forkids we discover on DNS...

I simply put some dbg in here and ran holesky node

Reply:
the test `test_get_node` calls `test_convert_enr_node_record` indirectly, with a real enr 

https://github.com/paradigmxyz/reth/blob/d6c50d776a92bf532434e328e69d9c98ed69bd1d/crates/net/dns/src/lib.rs#L462-L465

but I can add a unit test sure

ENRs received over a network are not guaranteed to have set the `"eth"` field with any particular type, we cannot assume all nodes will use it for fork id

Reply:
tbh, I really don't know what the expected format is, but decoding currently fails for

```
        (
            "eth",
            "c7c6849b192ad080",
        ),
```

though unclear if this is a valid forkid, also getting lots of invalid signature errors in enr decoding which makes me think the enr decoding impl is completely broken.

Reply:
https://github.com/sigp/enr/blob/df661bb47fff0e76d958b2cf73efa1e4f60918de/src/lib.rs#L1075

Reply:
added another test, decoding works

Reply:
[996b0ed](https://github.com/paradigmxyz/reth/pull/7564/commits/996b0edfb0a46a19d2e8a70f025530aee92aa831)

Reply:
yeah you are totally right, it's fudged. adding another field to the enr leads to broken decoding of the whole enr. this doesn't block the dns code, but it will fail for all enrs that have an extra field that is not a pre-defined key there in the encoding. this is a big bug since all enrs using `eth2` as key for kv pair (all cl nodes) will fail decoding. @mattsse